src/helics/cpp98/Federate.hpp

```diff
@@ -765,9 +829,56 @@ class Federate {
     bool exec_async_iterate;  //!< indicator that the federate is in an async operation
 #if defined(HELICS_HAS_FUNCTIONAL) && HELICS_HAS_FUNCTIONAL != 0
   private:
-    void* callbackBuffer{nullptr};  //!< buffer to contain pointer to a callback
-    void* timeUpdateCallbackBuffer{nullptr};  //!< buffer for pointer to time update callback
-    void* stateChangeCallbackBuffer{nullptr};  //!< buffer for pointer to state change callback
+    static constexpr int numberOfCallbacks = 5;
+    static constexpr int timeRequestEntryCallbackLocation = 0;
+    static constexpr int timeUpdateCallbackLocation = 1;
+    static constexpr int stateChangeCallbackLocation = 2;
+    static constexpr int queryCallbackLocation = 3;
+    static constexpr int timeRequestReturnCallbackLocation = 4;
+    void** callbackBuffers{nullptr};
```

Comment:
Does this need to be a pointer to an array of function pointers that need to get allocated, instead of a fixed size array of function pointers?